pyccl/tests/test_lpt_power.py

```diff
@@ -0,0 +1,171 @@
+import numpy as np
+import pyccl as ccl
+import pytest
+
+
+COSMO = ccl.Cosmology(
+    Omega_c=0.27, Omega_b=0.045, h=0.67, sigma8=0.8, n_s=0.96,
+    transfer_function='bbks', matter_power_spectrum='linear')
+TRS = {'TG': ccl.nl_pt.PTNumberCountsTracer(b1=2.0, b2=2.0, bs=2.0, bk2=2.0),
+       'TM': ccl.nl_pt.PTMatterTracer()}
+PTC = ccl.nl_pt.LagrangianPTCalculator(cosmo=COSMO)
+
+
+def test_lpt_calculator_smoke():
+    c = ccl.nl_pt.LagrangianPTCalculator(log10k_min=-3,
+                                         log10k_max=1,
+                                         nk_per_decade=10)
+    assert len(c.k_s) == 40
+
+
+@pytest.mark.parametrize('tr1,tr2',
+                         [['TG', 'TG'],
+                          ['TG', 'TM'],
+                          ['TM', 'TG'],
+                          ['TM', 'TM']])
+def test_lpt_get_pk2d_smoke(tr1, tr2):
+    t2 = None if tr2 == tr1 else TRS[tr2]
+    ptc = ccl.nl_pt.LagrangianPTCalculator(cosmo=COSMO)
+
+    pk = ptc.get_biased_pk2d(TRS[tr1], tracer2=t2)
+    assert isinstance(pk, ccl.Pk2D)
+
+
+@pytest.mark.parametrize('nl', ['nonlinear', 'linear', 'pt'])
+def test_lpt_get_pk2d_nl(nl):
+    ptc = ccl.nl_pt.LagrangianPTCalculator(
+        b1_pk_kind=nl, bk2_pk_kind=nl, cosmo=COSMO)
+    pk = ptc.get_biased_pk2d(TRS['TG'])
+    assert isinstance(pk, ccl.Pk2D)
+
+
+@pytest.mark.parametrize('typ_nlin,typ_nloc', [('linear', 'nonlinear'),
+                                               ('nonlinear', 'linear')])
+def test_lpt_k2pk_types(typ_nlin, typ_nloc):
+    tg = ccl.nl_pt.PTNumberCountsTracer(1., 0., 0., bk2=1.)
+    tm = ccl.nl_pt.PTNumberCountsTracer(1., 0., 0.)
+    ptc1 = ccl.nl_pt.LagrangianPTCalculator(
+        b1_pk_kind=typ_nlin, bk2_pk_kind=typ_nloc, cosmo=COSMO)
+    ptc2 = ccl.nl_pt.LagrangianPTCalculator(
+        b1_pk_kind=typ_nloc, cosmo=COSMO)
+    pkmm = ptc1.get_biased_pk2d(tm, tracer2=tm)
+    pkmm2 = ptc2.get_biased_pk2d(tm, tracer2=tm)
+    pkgg = ptc1.get_biased_pk2d(tg, tracer2=tg)
+    ks = np.geomspace(1E-3, 1E1, 128)
+    p1 = pkgg(ks, 1., cosmo=COSMO)
+    p2 = pkmm(ks, 1., cosmo=COSMO)+ks**2*pkmm2(ks, 1., cosmo=COSMO)
+    assert np.allclose(p1, p2, atol=0, rtol=1E-4)
+
+
+@pytest.mark.parametrize('kind', ['m:m', 'm:b2', 'm:b3nl', 'm:bs',
+                                  'b1:b3nl', 'b3nl:b3nl', 'b3nl:bk2'])
```

Comment:
Question: you only added these because testing all combinations takes a bit too long, right?

Reply:
I think that's OK, btw, just wanted to check.

Reply:
Yeah that's the thing driving the coverage. The problem is the 1, delta_L split in LPT. Because of that you can't isolate the bias terms as you can in EPT. I think these are the only ones that work.